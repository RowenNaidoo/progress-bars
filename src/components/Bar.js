import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Bar = ({ barValue }) => {
  return (
    <BarContainer>
      <UsageAmount id='barValue'>
        {barValue}%
      </UsageAmount>
      <BarFill barValue={barValue}>
      </BarFill>
    </BarContainer>
  )
}

const BarContainer = styled.div`
  width: 100%;
  height: 40px;
  border: 1px solid gray;
`

const BarFill = styled.div`
  width: ${props => props.barValue}%;
  max-width: 100%;
  height: 40px;
  background-color: ${props => props.barValue < 100 ? '#85C1E9' : 'red'};
  transition: all 0.2s ease;
`

const UsageAmount = styled.div`
  position: absolute;
  left: 50%;
  height: 40px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-weight: bold;
`

Bar.propTypes = {
  barValue: PropTypes.number.isRequired
}

export default Bar