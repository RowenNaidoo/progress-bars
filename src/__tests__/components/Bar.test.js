import React from 'react'
import { shallow } from 'enzyme'
import Bar from '../../components/Bar'


describe('#Bar', () => {
  const component = shallow(<Bar barValue={50} />)

  test('renders learn react link', () => {
    expect(component).toMatchSnapshot()
  })

  test('bar displays correct value', () => {
    expect(component.find('#barValue').text()).toEqual('50%')
  })

})