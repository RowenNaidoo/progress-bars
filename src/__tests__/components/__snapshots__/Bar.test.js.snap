// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Bar renders learn react link 1`] = `ShallowWrapper {}`;
